### PR TITLE
[Integration Tests] [BlockStoreSignaled] Additional tests

### PR DESCRIPTION
**Pull request by zeptin**
These tests primarily relate to detecting the presence or absence of messages getting exchanged between nodes.

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    public class BlockStoreSignaledTests
+    {
+        protected readonly ILoggerFactory loggerFactory;
+        private readonly Network network;
+
+        public BlockStoreSignaledTests()
+        {
+            this.loggerFactory = new LoggerFactory();
+
+            this.network = KnownNetworks.RegTest;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs]: Can you rather use new BitcoinRegTest here? We want to move away from the static container eventually.

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+        [Fact]
+        public void CheckBlocksAnnouncedAndQueueEmptiesOverTime()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode stratisNodeSync = builder.CreateStratisPowNode(this.network).NotInIBD().WithWallet();
+                CoreNode stratisNode1 = builder.CreateStratisPowNode(this.network).NotInIBD();
+
+                builder.StartAll();
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs]: You dont need the StartAll line anymore, you can just add .Start() at the end of the builder line, after WithWallet()

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                stratisNode1.CreateRPCClient().AddNode(stratisNodeSync.Endpoint, true);
+
+                INetworkPeer connectedPeer = node1ConnectionManager.ConnectedPeers.FindByEndpoint(stratisNodeSync.Endpoint);
+                TestBehavior testBehavior = connectedPeer.Behavior<TestBehavior>();
+
+                // We expect that node0 will abandon the 10 block chain and use the 15 block chain from node1
+                TestHelper.WaitLoop(() => stratisNode1.CreateRPCClient().GetBestBlockHash() == stratisNodeSync.CreateRPCClient().GetBestBlockHash());
+
+                // Connect all nodes together
+                stratisNode2.CreateRPCClient().AddNode(stratisNodeSync.Endpoint, true);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs]: Use TestHelper

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+                stratisNode1.CreateRPCClient().AddNode(stratisNode2.Endpoint, true);
+
+                INetworkPeer connectedPeer2 = node2ConnectionManager.ConnectedPeers.FindByEndpoint(stratisNodeSync.Endpoint);
+                TestBehavior testBehavior2 = connectedPeer2.Behavior<TestBehavior>();
+
+                // Wait for node2 to sync; it should have the 15 block chain
+                TestHelper.WaitLoop(() => stratisNode2.CreateRPCClient().GetBestBlockHash() == stratisNodeSync.CreateRPCClient().GetBestBlockHash());
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs]: Use TestHelper.AreNodesSynced.

File: src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+        {
+            this.loggerFactory = new LoggerFactory();
+
+            this.network = new BitcoinRegTest();
+            var serializer = new DBreezeSerializer();
+            serializer.Initialize(this.network);
+        }
+
+        [Fact]
+        public void CheckBlocksAnnouncedAndQueueEmptiesOverTime()
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStore/BlockStoreSignaledTests.cs]: Please add underscores "_" between logical separations for example
CheckBlocksAnnounced_AndQueueEmptiesOverTime

**Comment by dangershony** [comment]
@fassadlr please review or dismiss

**Pull request merged**